src/drivers/uavcannode/uavcannode_params.c

```diff
@@ -34,13 +34,15 @@
 /**
  * UAVCAN Node ID.
  *
+ * If 0, the node ID will be assigned by the network.
```

Comment:
IDR but I feel like a 0 value could be a problem. Please check the implications of a 0 node address.

Reply:
With these changes, a 0 node ID causes the dynamic node ID allocation to take place at runtime. This logic has always been present in **UavcanNode.cpp** but has never been reachable due to the default Node ID value of 120 which would cause it to [exit here](https://github.com/PX4/PX4-Autopilot/pull/19403/files#diff-bed8567c75487a628a8e116d6010cdd6ad7fa9d0b7e0240e99fbdbffedfcd1a0L636)